GitCommands/System.cs

```diff
@@ -245,4 +245,52 @@ public static string AsForce(this bool force)
             return force ? " -f " : string.Empty;
         }
     }
+
+    public static class DeconstructionExtensions
```

Comment:
.NET Core 2 supports deconstruction of (some of?) these types. For netfx, we must add extensions.

Reply:
Please move this class to its own file

Reply:
Done. `System.cs` already had two classes in there, so now there are three separate files.

This file had other horrors too, like the `System` namespace. GE namespaces could do with a review one day.